### PR TITLE
Tests documentation

### PR DESCRIPTION
**Pull request by shernanm**
## Summary

* The tests, and their respective documentation for the endpoint users, were implemented, taking into account the following cases:

  * Test when the creation of a user is satisfactory.
  * Test when the creation of a user fails to use an email that is in use.
  * Test when the creation of a user fails because the password does not meet the conditions.
  * Test when the creation of a user fails by not sending a mandatory parameter.

## Trello Card

https://trello.com/c/0VzDUC4G

### CODE REVIEW

**Comment by raparicio6** [comment]
Update with master

**Review by raparicio6 — APPROVED** [review]

File: test/users.test.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+const request = require('supertest');
+const dictum = require('dictum.js');
+
+const app = require('../app');
+
+const controller = request(app);
+
+describe('POST /users', () => {
+  test('Successful test when registering user', () =>
```
Comment (molinajulian) [inline test/users.test.js]: Replace `()` for `done` in all test.

File: test/users.test.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    controller
+      .post('/users')
+      .send({
+        firstName: 'John',
+        lastName: 'Katzenbach',
+        email: 'john.katz@wolox.co',
+        password: '12345678'
+      })
+      .then(response => {
+        expect(response.statusCode).toBe(201);
```
Comment (molinajulian) [inline test/users.test.js]: Search user in db and check fields.

File: test/users.test.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        return dictum.chai(response, 'Successful test creating user');
```
Comment (molinajulian) [inline test/users.test.js]: Change this for `dictum.chai(response)`

File: test/users.test.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        controller
+          .post('/users')
+          .send({
+            firstName: 'Jose',
+            lastName: 'Perez',
+            email: 'jose.perez@wolox.com.ar',
+            password: '12345test'
+          })
+          .then(response => {
+            expect(response.statusCode).toBe(503);
```
Comment (molinajulian) [inline test/users.test.js]: This is wrong. You must be return `400 BAD REQUEST` when the request contain an email exist. Change whatever is needed for that.

Comment (molinajulian) [reply test/users.test.js]: Also check the message for error

File: documentation.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+const dictum = require('dictum.js');
```
Comment (molinajulian) [inline documentation.js]: This file is unnecessary. When you call `dictum.chai(response)` the file with documentation is created automatically.

File: test/users.test.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+  test('User creation test when required parameters are not sent', () =>
+    controller
+      .post('/users')
+      .send({
+        email: 'john.katz@wolox.co',
+        password: '12345678'
+      })
+      .then(response => {
+        expect(response.statusCode).toBe(503);
```
Comment (molinajulian) [inline test/users.test.js]: This is a 400 bad request, no 503. 503 is bad gateway, and representing when the server is down.

**Review by molinajulian — CHANGES_REQUESTED** [review]

File: test/users.test.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+      .send({
+        firstName: 'John',
+        lastName: 'Katzenbach',
+        email: 'john.katz@wolox.co',
+        password: '12345678'
+      })
+      .then(response =>
+        User.findOne({
+          where: { email: 'john.katz@wolox.co' }
+        }).then(userDb => {
```
Comment (molinajulian) [inline test/users.test.js]: You can deconstruct here too :+1: . `({firstName,lastName,email})` instead of `userDb`

**Review by molinajulian — CHANGES_REQUESTED** [review]

**Review by raparicio6 — APPROVED** [review]

**Review by molinajulian — APPROVED** [review]

**Pull request merged**